single_cell_RNAseq/pipeline_pre_qc.nf

```diff
@@ -153,6 +155,13 @@ workflow {
                                             .map{it -> [it[0], it[6], it[2], it[3], it[4], it[5]]} // [lib, num_of_samples, plp_files]] (excluding .tsv)
             FREEMUXLET_POOL(ch_multi_lib_transformed)
 
+            if (params.settings.fmx_assign_to_gt){
```

Comment:
As intuitive as it may be to add this logic under the freemuxlet conditionals, I think conceptually and for readability it is better to put this after the de-multiplexing stage. That way we can leave the intention of block entirely for de-multiplexing and the outputs of de-multiplexing flow into the `FMX_ASSIGN_TO_GT` process. How do you feel about adding it on line 263 (after de-multiplexing). Something like:



```groovy
if (params.settings.demux_method == "freemuxlet" && params.settings.fmx_assign_to_gt) {
    if (params.settings.merge) {
        FMX_ASSIGN_TO_GT(
            Channel.from(get_pool_vcf())
                .join(FREEMUXLET_POOL.out.vcf) // [pool, ref_vcf, fmx_vcf]
        );
    }

    ch_gt_input = Channel.from(get_pool_vcf()) // [pool, vcf]
        .combine(
            Channel.from(get_library_by_pool()).map { it -> [it[1], it[0]] }, 
            by: 0
        ) // [pool, vcf, lib]
        .map { it -> [it[2], it[1]] } // [lib, vcf]
        .join(FREEMUXLET_LIBRARY.out.vcf); // [lib, ref_vcf, fmx_vcf]

    // TODO: add checks that the reference file exists
    FMX_ASSIGN_TO_GT_LIBRARY(ch_gt_input);
}

```

This also de-duplicates some of the repeated code for the single library processing

Reply:
Ooo good call. I'll move it around, thanks!